avocado_vt/plugins/vt_runner.py

```diff
@@ -1,53 +1,54 @@
-import logging
 import multiprocessing
 import os
-import sys
-import tempfile
 import time
 import traceback
 
-from avocado.core import exceptions, nrunner, output, teststatus
+from avocado.core import exceptions, nrunner, teststatus
 from avocado.core.runners.utils import messages
-from avocado.utils import astring, path, stacktrace
+from avocado.core.test_id import TestID
+from avocado.utils import astring
 
-from avocado_vt import utils
-from virttest import (data_dir, env_process, error_event, utils_env, utils_misc,
-                      utils_params, version, funcatexit)
+from avocado_vt import test
 
 
-class VirtTest(utils.TestUtils):
+class VirtTest(test.VirtTest):
 
     def __init__(self, queue, runnable):
-        self.__vt_params = utils_params.Params(runnable.kwargs)
         self.queue = queue
-        self.config = runnable.config
-        self.tmpdir = tempfile.mkdtemp()
-        self.logdir = os.path.join(self.tmpdir, 'results')
-        path.init_dir(self.logdir)
-        self.logfile = os.path.join(self.logdir, 'debug.log')
-        self.log = output.LOG_JOB
-        self.log_level = runnable.config.get('job.output.loglevel',
-                                             logging.DEBUG)
-        self.env_version = utils_env.get_env_version()
-        self.iteration = 0
-        self.background_errors = error_event.error_events_bus
-        # clear existing error events
-        self.background_errors.clear()
-        self.debugdir = self.logdir
-        self.bindir = data_dir.get_root_dir()
-        self.virtdir = os.path.join(self.bindir, 'shared')
-
-    @property
-    def params(self):
-        return self.__vt_params
-
-    def run_test(self):
-        status = "ERROR"
+        vt_params = runnable.kwargs
+        vt_params['job_env_cleanup'] = 'no'
```

Comment:
So as it stands (without the changes proposed here), running a `vt` test with `nrunner` means it already lacks the `funcatexit` feature, with the cleanup of the env, so I don't see this forceful override changing the current behavior.   Still, because we're logging the parameters, it may be a good idea to log a line about this (and the fact that env will not be clean up as in the traditional runner).

Reply:
@clebergnu Could you clarify what you mean by the "`funcatexit` feature" so that I can follow? We do have features on Avocado VT for instance where we want to keep the vms between runs, be it for debugging reasons or for greater reuse (as these could be large and bulky) so some settings like these have been introduced in some of my PR-s in the past too. If I understand what you mean, we simply don't need the settings since this is a default behavior for Avocado VT, correct?

Reply:
@pevogam avocado legacy runner has a `funcatexit` feature which is used to register methods to the legacy runner queue. Those methods are run at the end of avocado Job. The legacy avocado-vt is using this feature to clean and delete the environment after the testing. Unfortunately, this is not yet supported on the nrunner so the avocado-vt environment backing file is left in the system. But I think this is not problem for this PR, because it is not new for vt nrunner.